### DISCUSSION TITLE
Load fontsource font with Mantine

### DISCUSSION THREAD

**Discussion by Moshyfawn** [mantinedev/mantine#1162] [Q&A] [answered]
Hey, there!

I'm very impressed with Mantine UI and looking into refactoring my project to use it.

Can you, please, advise on how to load self-hosted fonts from [fontsource](https://github.com/fontsource/fontsource) to include them on a page.

I've tried importing the fontsource style sheet at the top of an app
```js
import "@fontsource/cabin"
```
and included the font family name into the `theme` object I pass into [the `MantineProvider`](https://mantine.dev/theming/mantine-provider/) _(most of the JSX omitted for readability)_
```jsx
const theme = {
  fontFamily: "Cabin, sans-serif"
}

return <MantineProvider theme={theme} />
```
This results in a system font loaded and displayed on the page instead.

I also tried prepending the styles as described in [the Mantine docs section about styles insertion point](https://mantine.dev/theming/mantine-provider/#change-styles-insertion-point) to no avail.

I'm open to suggestions, thx!

P.S. [Here's somewhat of a reproduction CSB](https://codesandbox.io/s/dazzling-curie-i1rtl8?file=%2Fsrc%2FApp.tsx)

**Comment by rtivital** [comment, marked as answer]
https://codesandbox.io/s/busy-sun-lvowu3?file=/src/App.tsx

**Reply by Moshyfawn** [reply]
Inspecting the heading element, it's reported to display the Helvetica font, huh

<img width="1440" alt="image" src="https://user-images.githubusercontent.com/16290753/161381073-22dc86f3-38e3-4eb1-b6c7-aeb173d34354.png">

Looking in the console, there's an error
```bash
downloadable font: rejected by sanitizer (font-family: "Cabin" style:normal weight:400 stretch:100 src index:0) source: https://lvowu3.csb.app/files/cabin-latin-400-normal.woff2
```

IDK, if it's Mantine UI or smth else, TBH. Using [MUI](mui.com/), I haven't encountered this issue before

**Reply by rtivital** [reply]
font-family is set correctly for headings and other elements on Mantine side, everything else is managed on your side.

**Reply by Moshyfawn** [reply]
Also, is it by design that setting `fontFamily` doesn't propagate to all the headings or any other text elements?

Is it expected to provide the `fontFamily` name to every other text component having `fontFamily` defined at the `root` of the theme object?
```diff
const theme = {
  fontFamily: "Cabin, sans-serif",
+ headings: { fontFamily: "Cabin, sans-serif" }
};
```

**Reply by rtivital** [reply]
You can find everything about typography in documentation – https://mantine.dev/theming/extend-theme/#typography

**Reply by Moshyfawn** [reply]
> You can find everything about typography in documentation – https://mantine.dev/theming/extend-theme/#typography

Ah, missed this remark
>- `theme.fontFamily` – controls font-family in all other components **except headings**

My bad, thx!

**Reply by Moshyfawn** [reply]
> font-family is set correctly for headings and other elements on Mantine side, everything else is managed on your side.

So, the issue is somewhere in [the CSB](https://codesandbox.io/s/busy-sun-lvowu3?file=/src/App.tsx) as it works fine locally in a Vite project.

Thank you, @rtivital! You've immensely helped me with my inattentiveness :P

**Reply by pencilcheck** [reply]
URL has been updated to: https://mantine.dev/theming/typography/
